File: python/BinaryGap.py
def solution(N):
    maxgap = 0
    tmpgap = -1
    n = 1

    for i in range(1,32):
        if n & N:
            maxgap = max(maxgap, tmpgap)
            tmpgap = 0
        elif tmpgap != -1:
            tmpgap += 1
            
        n <<= 1

    return maxgap

File: cpp/BinaryGap.cpp
int solution(int N) {

    int mask = 1;
    int maxgap = 0;
    int tmpgap = -1;
    for(int i = 1; i<33; i++)
    {
        if(mask & N) 
        {
            maxgap = max(maxgap, tmpgap);
            tmpgap = 0;
        }
        else if(tmpgap!=-1)
        {
            tmpgap++;
        }
        mask <<= 1;
    }

    return maxgap;
}
